x-pack/legacy/plugins/siem/public/containers/detection_engine/rules/fetch_index_patterns.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { isEmpty, get } from 'lodash/fp';
+import { useEffect, useState, Dispatch, SetStateAction } from 'react';
+import { StaticIndexPattern } from 'ui/index_patterns';
+
+import { getIndexFields, sourceQuery } from '../../../containers/source';
+import { useStateToaster } from '../../../components/toasters';
+import { errorToToaster } from '../../../components/ml/api/error_to_toaster';
+import { SourceQuery } from '../../../graphql/types';
+import { useApolloClient } from '../../../utils/apollo_context';
+
+import * as i18n from './translations';
+
+interface FetchIndexPattern {
+  isLoading: boolean;
+  indices: string[];
+  indicesExists: boolean;
+  indexPatterns: StaticIndexPattern | null;
+}
+
+type Return = [FetchIndexPattern, Dispatch<SetStateAction<string[]>>];
+
+export const useFetchIndexPatterns = (): Return => {
+  const apolloClient = useApolloClient();
+  const [indices, setIndices] = useState<string[]>([]);
+  const [indicesExists, setIndicesExists] = useState(false);
+  const [indexPatterns, setIndexPatterns] = useState<StaticIndexPattern | null>(null);
```

Comment:
Nit: The `StaticIndexPattern | null` is redundant I think.

But optional on this one.